### DISCUSSION TITLE
mobile responsive design

### DISCUSSION THREAD

**Discussion by legaljustus** [Budibase/budibase#1651] [Ideas]
Hi!

I was wondering if there is a way to easily make my app more mobile responsive. 
I tried adding breakpoints using css with [this method](https://www.w3schools.com/css/css_rwd_mediaqueries.asp).
But I think that's not possible with the current way budibase handles custom css.
I tried a lot of different layout combinations (flex, wrap etc. however i'm certainly not experienced with css layout) and I can't seem to get it to get it the way I want. I think that adding something like the wp page builder oxygen has for custom mobile responsive layouts could be really useful, you can customize the design for each device / screen size:

![image](https://user-images.githubusercontent.com/83215912/121013955-bbd2a500-c799-11eb-9c63-5142b8cf6614.png)

Or maybe I just need to look into css layout a bit more : )

**Comment by mjashanks** [comment]
This is coming at the end of June :)

We will allow fixed breakpoints for mobile, tablet and desktop.

**Reply by Kuuchuu** [reply]
Hey @mjashanks, just wanted to see if this feature was added, and if so how would one utilize it? I didn't see anywhere in the docs. Would it be classified as custom CSS? Thanks

**Reply by mjashanks** [reply]
@Kuuchuu  This was done!

In the design preview, you can change screen size:

![image](https://user-images.githubusercontent.com/3524181/148121913-7cd14ecc-3e52-4f29-902c-00aa699eafcc.png)

And you can modify how things look using Conditional UI + the bindings `{{ Device.Tablet }}` and `{{ Device.Mobile }}`.

Also, many of our components were made to be responsive by default - especially the "Blocks"
